node/src/components/deploy_buffer/tests.rs

```diff
@@ -0,0 +1,590 @@
+    reactor::{EventQueueHandle, QueueKind, Scheduler},
+    types::{Block, FinalizedBlock},
+    utils,
+};
+use casper_types::{testing::TestRng, EraId, ProtocolVersion, TimeDiff};
+use prometheus::Registry;
+use rand::Rng;
+
+enum DeployType {
+    NativeTransfer,
+    Contract,
+    Random,
+}
+
+// Generates valid deploys
+fn create_valid_deploys(
+    rng: &mut TestRng,
+    size: usize,
+    deploy_type: DeployType,
+    strict_timestamp: Option<Timestamp>,
+    with_ttl: Option<TimeDiff>,
+) -> Vec<Deploy> {
+    let mut deploys = Vec::with_capacity(size);
+
+    for _ in 0..size {
+        let deploy_ttl = match with_ttl {
+            Some(ttl) => ttl,
+            None => TimeDiff::from_seconds(rng.gen_range(30..100)),
+        };
+        let deploy_timestamp = match strict_timestamp {
+            Some(timestamp) => timestamp,
+            None => Timestamp::now(),
+        };
+        match deploy_type {
+            DeployType::NativeTransfer => {
+                deploys.push(Deploy::random_valid_native_transfer_with_timestamp_and_ttl(
+                    rng,
+                    deploy_timestamp,
+                    deploy_ttl,
+                ));
+            }
+            DeployType::Contract => {
+                if strict_timestamp.is_some() {
+                    unimplemented!();
+                }
+                let deploy = Deploy::random_with_valid_session_package_by_name(rng);
+                assert!(deploy.is_valid().is_ok());
+                deploys.push(deploy);
+            }
+            DeployType::Random => {
+                deploys.push(Deploy::random_with_timestamp_and_ttl(
+                    rng,
+                    deploy_timestamp,
+                    deploy_ttl,
+                ));
+            }
+        }
+    }
+
+    deploys
+}
+
+fn create_invalid_deploys(rng: &mut TestRng, size: usize) -> Vec<Deploy> {
+    let mut deploys = create_valid_deploys(rng, size, DeployType::Random, None, None);
+
+    for deploy in deploys.iter_mut() {
+        deploy.invalidate();
+    }
+
+    deploys
+}
+
+// Checks sizes of the deploy_buffer containers. Also checks the metrics recorded.
+fn assert_container_sizes(
+    deploy_buffer: &DeployBuffer,
+    expected_buffer: usize,
+    expected_dead: usize,
+    expected_held: usize,
+) {
+    assert_eq!(deploy_buffer.buffer.len(), expected_buffer);
+    assert_eq!(deploy_buffer.dead.len(), expected_dead);
+    assert_eq!(
+        deploy_buffer
+            .hold
+            .values()
+            .map(|deploys| deploys.len())
+            .sum::<usize>(),
+        expected_held
+    );
+    assert_eq!(
+        deploy_buffer.metrics.total_deploys.get(),
+        expected_buffer as i64
+    );
+    assert_eq!(
+        deploy_buffer.metrics.held_deploys.get(),
+        expected_held as i64
+    );
+    assert_eq!(
+        deploy_buffer.metrics.dead_deploys.get(),
+        expected_dead as i64
+    );
+}
+
+#[test]
+fn test_register_deploy() {
```

Comment:
For consistent naming, we shouldn't have test names beginning with "test". Instead, test names usually start with either the component name followed by the function it's testing or the function it's testing followed by the expected action.

E.g. "block_synchronizer_purge_historical", "need_next_invalid_state_should_fail"

Applies to all tests in this PR.

Reply:
changed here: https://github.com/casper-network/casper-node/pull/3544/commits/3235906ceeb460ac3160bb23ccff0125ed409184